src/operations/compose-filters.ts

```diff
@@ -0,0 +1,15 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+export type Predicate<T> = (item: T) => boolean;
+
+export function composeFilters<T>(...predicates: Array<null | Predicate<T>>): null | Predicate<T> {
+  let composed: null | Predicate<T> = null;
+  for (const predicate of predicates) {
+    if (predicate) {
+      const previous: Predicate<T> = composed ?? (() => true);
+      composed = (item: T) => previous(item) && predicate(item);
```

Comment:
Fine when you have only 2 filters, but you introduced a potential for call stack overflow with this recursion, along with performance concerns.

Why not simply like this:

```js
function composeFilters(..predicates) {
   return (item) => {
      for(const predicate of predicates) {
           if(!predicate(item)) {
               return false;
           }
      }
      return true;
   } 
}
```

Reply:
I doubt there is ever going to be as many filters but I don't mind the suggested approach. I tested both before implementation and the one I've chosen is ever so slightly faster, but the difference is tiny.